kernel/sched.c

```diff
@@ -158,8 +158,10 @@ static inline bool is_halting(struct k_thread *thread)
 /* Clear the halting bits (_THREAD_ABORTING and _THREAD_SUSPENDING) */
 static inline void clear_halting(struct k_thread *thread)
 {
+#if CONFIG_MP_MAX_NUM_CPUS > 1
```

Comment:
Style: IMHO a regular `if()` would be cleaner and clearer here.

Reply:
Non-style: CONFIG_SMP is the better tunable to test here, not the number of cores.  Non-SMP multicore builds are a legal edge case, and you can only run Zephyr threads on one core, so it doesn't need this.

Reply:
As I understand it, CONFIG_SMP with but 1 core is also legit. Perhaps something like ...

`#if defined(CONFIG_SMP) && (CONFIG_MP_MAX_NUM_CPUS > 1)`

... or its `if ()` equalivent.